frontend/src/components/datasets/ShareGroupDatasetModal.tsx

```diff
@@ -0,0 +1,110 @@
+import {fetchGroups} from "../../actions/group";
+import {RootState} from "../../types/data";
+import {parseDate} from "../../utils/common";
+
+
+type ShareGroupDatasetModalProps = {
+    open: boolean,
+    handleClose: any,
+    datasetName: string
+}
+
+export default function ShareGroupDatasetModal(props: ShareGroupDatasetModalProps) {
+    const { open, handleClose, datasetName } = props;
+
+    const [group, setGroup] = useState("")
+    const [role, setRole] = useState("viewer")
+	const dispatch = useDispatch();
+	const listGroups = () => dispatch(fetchGroups(0, 21));
+	const groups = useSelector((state: RootState) => state.group.groups);
+
+	// component did mount
+	useEffect(() => {
+		listGroups();
+	}, []);
+
+    const onShare = () => {
+        handleClose();
+    }
+	console.log('groups are here', groups);
+
+
+	const options = [];
+    groups.map((group) => {
+    	let group_option = {value:group.id, label:group.name}
+		group_options.push(group_option);
+    });
+	console.log('group optioins are', options);
+	console.log('it is of type', typeof(options));
+	const initial_group = options.at(0);
+	console.log('initial group is', initial_group);
+    return (
+        <Container>
+            <Dialog open={open} onClose={handleClose} fullWidth={true} maxWidth="md"
+                sx={{
+                    '.MuiPaper-root': {
+                        padding: "2em",
+                    },
+                }}>
+                <DialogTitle>Share dataset &apos;{datasetName}&apos;</DialogTitle>
+                <Divider />
+                <DialogContent>
+                    <Typography>Invite groups to collaborate</Typography>
+                    <div style={{
+                        display: 'flex',
+                        alignItems: 'center'
+                    }}>
+                        <Autocomplete
+                            id="email-auto-complete"
+                            freeSolo
+                            autoHighlight
+                            inputValue={group}
+                            onInputChange={(event, value) => {
+                                setGroup(value)
+                            }}
+                            options={["abc@xyz.com"]}
+                            renderInput={(params) => <TextField {...params} sx={{ mt: 1, mr: 1, "alignItems": "right", "width": "450px" }} required label="Enter email address" />}
+                        /> as
+						<Select
+								options={options}
+                                labelId="demo-simple-select-label"
+                                id="demo-simple-select"
+                                label="Group"
+								defaultGroup={initial_group}
+								value={group}
+                                onChange={(event, value) => {
+                                    setGroup(event.target.value);
+                                }}
+                            >
+                            </Select>
```

Comment:
This dropdown is always blank even though options shows all the groups added. Not sure what is wrong with it.